File: mobile/lib/models/wallet_transaction.dart
class WalletTransaction {
  final String id;
  final String walletId;
  final String transactionType;
  final int amountPaise;
  final String? referenceType;
  final String? referenceId;
  final String? description;
  final String createdBy;
  final String createdAt;

  WalletTransaction({
    required this.id,
    required this.walletId,
    required this.transactionType,
    required this.amountPaise,
    this.referenceType,
    this.referenceId,
    this.description,
    required this.createdBy,
    required this.createdAt,
  });

  factory WalletTransaction.fromJson(
    Map<String, dynamic> json,
  ) {
    return WalletTransaction(
      id: json['id']?.toString() ?? '',
      walletId: json['wallet_id']?.toString() ?? '',
      transactionType:
          json['transaction_type']?.toString() ?? 'UNKNOWN',
      amountPaise: json['amount_paise'] ?? 0,
      referenceType:
          json['reference_type']?.toString(),
      referenceId:
          json['reference_id']?.toString(),
      description:
          json['description']?.toString(),
      createdBy:
          json['created_by']?.toString() ?? '',
      createdAt:
          json['created_at']?.toString() ?? '',
    );
  }

  double get amount => amountPaise / 100;
}

File: backend/app/models/wallet_transaction.py
import uuid

from datetime import datetime

from sqlalchemy import (
    BigInteger,
    DateTime,
    ForeignKey,
    String,
    Text,
)
from sqlalchemy.orm import Mapped, mapped_column

from app.db.database import Base


class WalletTransaction(Base):

    __tablename__ = "wallet_transactions"

    id: Mapped[uuid.UUID] = mapped_column(
        primary_key=True,
        default=uuid.uuid4,
    )

    wallet_id: Mapped[uuid.UUID] = mapped_column(
        ForeignKey("wallets.id", ondelete="CASCADE"),
        nullable=False,
    )

    transaction_type: Mapped[str] = mapped_column(
        String(30),
        nullable=False,
    )

    amount_paise: Mapped[int] = mapped_column(
        BigInteger,
        nullable=False,
    )

    reference_type: Mapped[str | None] = mapped_column(
        String(30),
        nullable=True,
    )

    reference_id: Mapped[uuid.UUID | None] = mapped_column(
        nullable=True,
    )

    description: Mapped[str | None] = mapped_column(
        Text,
        nullable=True,
    )

    created_by: Mapped[uuid.UUID] = mapped_column(
        ForeignKey("users.id"),
        nullable=False,
    )

    created_at: Mapped[datetime] = mapped_column(
        DateTime,
        default=datetime.utcnow,
        nullable=False,
    )